### PR TITLE
Simplify and generalize switch statements in EXE backend

### PR DESCRIPTION
**Pull request by laurenthuberdeau**
## Context

Shorter and more general code for switch statements 🎉

### CODE REVIEW

File: exe.c [hunk 1]
```diff
@@ -101,6 +117,13 @@ void jump_rel(int offset);
 void call(int lbl);
 void ret();
 
+void dup(int reg) {
+  pop_reg(reg);
```
Comment (SamuelYvon) [inline exe.c:121]: How does the register system work in pnut? I haven't looked at it yet. Is the `int` just a mapping to an `enum`?

File: exe.c [hunk 2]
```diff
@@ -468,13 +475,13 @@ void cgc_add_enclosing_loop(int loop_fs, int break_lbl, ast continue_lbl) {
...
 
-void cgc_add_enclosing_switch(int loop_fs, int break_lbl) {
+void cgc_add_enclosing_switch(int loop_fs, int break_lbl, int next_case_lbl) {
   int binding = alloc_obj(5);
   heap[binding+0] = cgc_locals;
   heap[binding+1] = BINDING_SWITCH;
   heap[binding+2] = loop_fs;
   heap[binding+3] = break_lbl;
-  heap[binding+4] = 0;
+  heap[binding+4] = next_case_lbl;
```
Comment (SamuelYvon) [inline exe.c:484]: Why was this 0 before?

**Review by SamuelYvon — COMMENTED** [review]
LGTM, questions about the design of the register allocator

File: exe.c (hunk 2 above)
Comment (laurenthuberdeau) [reply exe.c:484]: I had the same question. I tried removing it and shrinking the object and it still worked so I think it was left by accident in a previous PR.

File: exe.c (hunk 1 above)
Comment (laurenthuberdeau) [reply exe.c:121]: The exe backend uses a stack machine so we don't need to do register allocation. Still, the exe backend declares 3 general purpose registers, `regX`, `regY` and `regZ`, [represented using ints](https://github.com/udem-dlteam/pnut/blob/master/exe.c#L87-L91) and [defined in the x86 file](https://github.com/udem-dlteam/pnut/blob/master/x86.c#L21-L46).

**Review by SamuelYvon — APPROVED** [review]
LGTM

**Pull request merged**